guide/english/algorithms/sieve-of-eratosthenes/index.md

```diff
@@ -0,0 +1,81 @@
+---
+title: sieve of Eratosthenes
+---
+
+## sieve of Eratosthenes
+Simple sieve algorithm for finding all prime numbers up to a designated limit.
+The earliest known reference to the sieve attributes it to Eratosthenes of
+Cyrene, a Greek mathematician.
+
+### How it works
+
+The algorithm operates on a set range of numbers starting at 1 through 'n'. It uses the following steps:
+
+  1. Create a list of all numbers in your range.
+  2. Remove 1 from the list, as it is not a prime number.
+  3. Circle the next number on the list.
+  4. Remove all multiples of the most recently circled number from the list.
+  5. Repeat steps 3 & 4 until you no longer have any numbers left in the list.
+
+  The first couple iterations look like this:
+  1.  * Circle 2
+      * Remove all multiples of 2 --- 4, 6, 8, ... 100
+  2.  * Circle 3
+      * Remove all multiples of 3 remaining on the list --- 9, 15, 21, ... 99
+
+### Example implementation in C
+
+C code: In this example our range is from 1 to n, where n == 100.
+
+```c
+
+#include <stdio.h>
+#include <stdlib.h>
+
+int main() {
+
+    int i, j;           // array indices
+    int *prime_array;   // array to mark if index is a prime number
+    int max = 100;      // max number to be tested
+
+    prime_array = malloc(sizeof(int) * max);
+
+    // explicitly stating that 0 & 1 are not prime numbers
+    // implicitly false otherwise
+    prime_array[0] = 0;
+    prime_array[1] = 0;
```

Comment:
If you are considering 0 as well, then remember that declaring an array with the size of **`100`**, means entries from `0 to 99`.
therefore if 0 is being considered, then the size of the array should be **`101`** and not **`100`**, so that entries could range from `0 to 100`